prompt-service/src/unstract/prompt_service/variable_extractor/prompt_variable_service.py

```diff
@@ -44,10 +45,21 @@ def check_static_variable_run_status(
             return output
 
     @staticmethod
-    def replace_generic_string_value(prompt: str, variable: str, value: str) -> str:
-        replaced_prompt = prompt.replace(variable, value)
+    def replace_generic_string_value(prompt: str, variable: str, value: Any) -> str:
+        formatted_value: str = value
+        if not isinstance(value, str):
+            formatted_value = VariableService.handle_json_and_str_types(value)
+        replaced_prompt = prompt.replace(variable, formatted_value)
         return replaced_prompt
 
+    @staticmethod
+    def handle_json_and_str_types(value: Any) -> str:
+        try:
+            formatted_value = json.dumps(value)
+        except ValueError:
```

Comment:
@harini-venkataraman shouldn't this be `JSONDecodeError`?

Reply:
Yes @chandrasekharan-zipstack Like we discussed, ValueError is parent of `JSONDecodeError`, Owing to priority, let us merge this. Will pick it up in next PR.